core/src/main/java/fi/iki/express/NanoExpress.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    public synchronized void addMappings(String path, Router route) {
+        //If a URI Path is already specified, it will be over written.
+        //It will rejoin the queue at the end.
+        if (path != null && route != null ) {
+            if ( this.route_priority.contains(path)) {
+                this.route_priority.remove(path);
+            }
+            this.route_priority.add(path);
+            this.routerArray.put(path, route);
+        }
+    }
+
+    /**
+     * User Define the mechanism of how he/she wants to load the mapping.
+     * Externally
+     */
+    public abstract void loadMappings();
+
+    /**
+     * The server will serves content based on the added routes.
+     * The server will serve the first route that matches and return a non null response.
+     *
+     * User can either programmatically add the mapping using one of the constructor or define the @loadMappings method to
+     * instruct the server how to load that mapping.
+     *
+     * Implementor of Router Interface will have full control of what will be served out from NanoHTTPD.
+     *
+     * <p/>
+     * (By default, this returns a 404 "Not Found" plain text error response.)
+     *
+     * @param session The HTTP session
+     * @return HTTP response, see class Response for details
+     */
+    @Override
+    public NanoHTTPD.Response serve(IHTTPSession session) {
+        Method method = session.getMethod();
+        NanoHTTPD.Response res = null;
+        try {
+            for (String uri : route_priority) {
+                Router r = routerArray.get(uri);
+                //This will follow the methods defined in NanoHTTPD.Method
+                switch (method) {
+                    case GET:
+                        res = r.doGet(session);
+                        break;
+                    case PUT:
+                        res = r.doPut(session);
+                        break;
+                    case POST:
+                        res = r.doPost(session);
+                        break;
+                    case DELETE:
+                        res = r.doDelete(session);
+                        break;
+                    case HEAD:
+                        res = r.doHead(session);
+                        break;
+                    case OPTIONS:
+                        res = r.doOptions(session);
+                        break;
+                    case TRACE:
+                        res = r.doTrace(session);
+                        break;
+                    case CONNECT:
+                        res = r.doConnect(session);
+                        break;
+                    case PATCH:
+                        res = r.doPatch(session);
+                        break;
+                    default:
+                        res = new IStatusResponse(Response.Status.METHOD_NOT_ALLOWED);
+                }
+                if (res != null) return res;
+            }
+        } catch (Exception e) {
```

Comment:
And at this place we lose the original exception. May be it would be better to place it as an argument of IStatusResponse (add constructor with some arguments). It is more frendly for user/developer. And, of course, print stac trace to console.